architecture/src/iosMain/kotlin/navigation/NavigationSpec.kt

```diff
@@ -0,0 +1,359 @@
+import platform.UIKit.UIDocumentBrowserUserInterfaceStyle
+import platform.UIKit.UIDocumentBrowserUserInterfaceStyleLight
+import platform.UIKit.UIDocumentBrowserViewController
+import platform.UIKit.UIDocumentBrowserViewControllerDelegateProtocol
+import platform.UIKit.UIImage
+import platform.UIKit.UIImagePickerController
+import platform.UIKit.UIImagePickerControllerDelegateProtocol
+import platform.UIKit.UIImagePickerControllerSourceType
+import platform.UIKit.UIInterfaceOrientation
+import platform.UIKit.UIInterfaceOrientationMask
+import platform.UIKit.UIModalPresentationAutomatic
+import platform.UIKit.UIModalPresentationStyle
+import platform.UIKit.UIModalTransitionStyle
+import platform.UIKit.UIModalTransitionStyleCoverVertical
+import platform.UIKit.UINavigationController
+import platform.UIKit.UINavigationControllerDelegateProtocol
+import platform.UIKit.UINavigationControllerOperation
+import platform.UIKit.UIView
+import platform.UIKit.UIViewController
+import platform.UIKit.UIViewControllerAnimatedTransitioningProtocol
+import platform.UIKit.UIViewControllerInteractiveTransitioningProtocol
+import platform.darwin.NSObject
+
+/**
+ * Spec used by [Navigator] to determine how to perform Navigation
+ */
+sealed class NavigationSpec {
+
+    /**
+     * Navigates to a new view controller by pushing it on top of the parents [UINavigationController]
+     * @param animated Specifies whether transition should be animated
+     * @param push Function to create the [UIViewController] to push
+     */
+    data class Push(val animated: Boolean = true, val push: () -> UIViewController) : NavigationSpec()
+
+    /**
+     * Pops the viewController on the parents [UINavigationController].
+     * @param to Optional [UIViewController] to pop to. If not provided the viewcontroller on top is popped.
+     * @param animated Specifies whether transition should be animated
+     */
+    data class Pop(val to: UIViewController? = null, val animated: Boolean = true) : NavigationSpec()
+
+    /**
+     * Lets the parent present a [UIViewController] using [UIViewController.presentViewController].
+     * @param animated Specifies whether transition should be animated
+     * @param presentationStyle The [UIModalPresentationStyle] to present the viewcontroller in. Defaults to [UIModalPresentationAutomatic]
+     * @param transitionStyle The [UIModalTransitionStyle] to transition to. Defaults to [UIModalTransitionStyleCoverVertical]
+     * @param present Function to create the [UIViewController] to present
+     * @param complete Optional function to call when presentation has completed
+     */
+    data class Present(
+        val animated: Boolean = true,
+        val presentationStyle: UIModalPresentationStyle = UIModalPresentationAutomatic,
+        val transitionStyle: UIModalTransitionStyle = UIModalTransitionStyleCoverVertical,
+        val present: () -> UIViewController,
+        val complete: (() -> Unit)? = null
```

Comment:
I'd name it `completion` to follow iOS style

Reply:
Renamed to completion